src/screens/AddNewItemScreen/index.tsx

```diff
@@ -13,32 +13,57 @@ import { style } from './style';
 import 'react-native-gesture-handler';
 import { createAlertMessageHelper } from '../../helpers/createAlertMessageHelper';
 import { AddNewItemScreeBackButton } from '../../components/AddNewItemScreenBackButton';
-import { useNavigation } from '@react-navigation/native';
-import { AddNewItemScreenNavigationProps } from './type';
+import { useNavigation, useRoute } from '@react-navigation/native';
+import {
+  AddNewItemScreenNavigationProps,
+  AddNewItemScreenRouteProps,
+} from './type';
+import { editItemSagaAction } from '../../redux/actions/todoSagaActions/editItemSagaAction';
+import { createCurrentDateHelper } from '../../helpers/createCurrentDateHelper';
+import { stopNotificationHelper } from '../../helpers/stopNotificationHelper';
 
 export const AddNewItemScreen: FC = () => {
   const navigation = useNavigation<AddNewItemScreenNavigationProps>();
+  const route = useRoute<AddNewItemScreenRouteProps>();
   const dispatch = useDispatch();
 
+  const isEdit = route.params.isEdit;
+  const editItem = route.params.editItem;
+  const initialState = editItem?.text ?? '';
+
+  useEffect(() => {
+    setDate(editItem?.notificationDate ?? new Date());
+    const title = isEdit ? 'Edit' : 'Add new task';
+    navigation.setOptions({ title: title });
+  }, []);
+
   const userToken = useSelector(userTokenSelector);
   const channelId = useSelector(deviceTokenSelector);
 
-  const [text, setText] = useState('');
+  const [text, setText] = useState(initialState);
   const [date, setDate] = useState(new Date());
   const [open, setOpen] = useState<boolean>(false);
-  const currentDate = `${date.getDate()}.${date.getMonth()}.${date.getFullYear()} at ${date.getHours()}:${date.getMinutes()}`;
-  const buttonStyle = text ? style.button : style.buttonDis;
+
+  const callback = () => {
+    navigation.goBack();
+  };
 
   const hasUnsavedChanges = !!text;
 
+  const currentDate = createCurrentDateHelper(date);
+  const buttonStyle = text ? style.button : style.buttonDis;
+
   const addItem = (text: TodoItemType['text']) => {
-    const newItem = createNewItemHelper(text);
+    const newItem = createNewItemHelper(text, date);
     createNotificationHelper({ channelId, newItem, date });
-    dispatch(addItemSagaAction({ newItem, userToken }));
+    dispatch(addItemSagaAction({ newItem, userToken, callback }));
   };
 
   const onPressBack = useCallback(() => {
     const back = () => {
+      navigation.setParams({
```

Comment:
why do you set this params on goback action?

Reply:
because I have to return the original value of the parameter to render the add screen